packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -611,9 +614,6 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     if (searchText.isEmpty) {
       return null;
     }
-    if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
-      return currentHighlight;
-    }
```

Comment:
Just a question, why do we remove this block of code? I remember this part is added to fix the keyboard navigation: https://github.com/flutter/flutter/pull/147294.

Reply:
https://github.com/flutter/flutter/pull/147294  only works on the default search, so the bug still exist when a custom search callback is provided. I used another method to fix the issue which doesn't rely on the search function, and this code is no longer needed. 

The main reason for reverting https://github.com/flutter/flutter/pull/147294 was to keep default search consistent with `searchCallback` where it doesn't have direct access to `currentHighlight`. And since we don't know the match condition of a custom `searchCallback`, we cannot implement "prioritising current selection" for a `searchCallback`, the only way to keep it consistent is to remove the feature. Also I think the original intent of search is to be agnostic about the state of `DropdownMenu`.

However prioritising current selection is a bit more intuitive, and now https://github.com/flutter/flutter/pull/147294 has landed in stable, I am no longer sure removing the feature is correct, since it also affect the behaviour of the search.